client/my-sites/customer-home/cards/features/help-search/search-card.jsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { identity, includes } from 'lodash';
+import debugFactory from 'debug';
+
+/**
+ * Internal Dependencies
+ */
+import { recordTracksEvent } from 'state/analytics/actions';
+import SearchCard from 'components/search-card';
+import {
+	getInlineHelpCurrentlySelectedLink,
+	getSelectedResultIndex,
+	isRequestingInlineHelpSearchResultsForQuery,
+	getInlineHelpCurrentlySelectedResult,
+} from 'state/inline-help/selectors';
+import {
+	requestInlineHelpSearchResults,
+	selectNextResult,
+	selectPreviousResult,
+} from 'state/inline-help/actions';
+
+/**
+ * Module variables
+ */
+const debug = debugFactory( 'calypso:customer-home-help' );
+
+class HelpSearchCard extends Component {
+	static propTypes = {
+		openResult: PropTypes.func.isRequired,
+		translate: PropTypes.func,
+		query: PropTypes.string,
+	};
+
+	static defaultProps = {
+		translate: identity,
+		query: '',
+	};
+
+	onKeyDown = ( event ) => {
+		// ignore keyboard access when manipulating a text selection in input etc.
+		if ( event.getModifierState( 'Shift' ) ) {
+			return;
+		}
+		// take over control if and only if it's one of our keys
+		if ( includes( [ 'ArrowUp', 'ArrowDown', 'Enter' ], event.key ) ) {
+			event.preventDefault();
+		} else {
+			return;
+		}
+
+		switch ( event.key ) {
+			case 'ArrowUp':
+				this.props.selectPreviousResult();
+				break;
+			case 'ArrowDown':
+				this.props.selectNextResult();
+				break;
+			case 'Enter': {
+				const hasSelection = this.props.selectedResultIndex >= 0;
+				hasSelection && this.props.openResult( event, this.props.selectedResult );
+				break;
+			}
+		}
+	};
+
+	onSearch = ( searchQuery ) => {
+		debug( 'search query received: ', searchQuery );
+		this.props.recordTracksEvent( 'calypso_inlinehelp_search', {
```

Comment:
> Customer home tracks events are prefixed with calypso_customer_home and we usually also bump interesting stats events under the same grouping.

So we've been going around the houses about this naming convention. Essentially @davemart-in wants to be able to compare the stats for "support search" and tell whether it came from:

1. FAB 
2. My Home.

Therefore, whilst we're aware of the `calypso_customer_home` convention, we decided we'd be better off simply adding a new property to Tracks for the existing `calypso_inlinehelp_` event namespace.

eg:

```js
this.props.recordTracksEvent( 'calypso_inlinehelp_show', { 
    location: 'inline-help-popover' // new property
} );
```

and

```js
this.props.recordTracksEvent( 'calypso_inlinehelp_show', { 
    location: 'my-home-feature-help-search' // new property
} );
```

We could track the same data in parralell under the `calypso_customer_home_` namespace if folks think that would be useful.

To be honest I think the tracking side of this task is growing ever more complex and we need to step back and plan it out carefully before proceeding with enabling this feature.

cc @retrofox who was taking up the tracking side.

Reply:
what we'd like to get in terms of tracking is a way to compare the stats between these two components. FAB and `<Support />`

After talking with data folks, we determine that we should keep the same events and adding a new `location` property to the events in order to segment them. That's what Dave mention in the comment above:

```es6
// InlineHelpPopover
this.props.recordTracksEvent( 'calypso_inlinehelp_show', { 
    location: 'inline-help-popover' // new property
} );

// Support
this.props.recordTracksEvent( 'calypso_inlinehelp_show', { 
    location: 'my-home-feature-help-search' // new property
} );
```

But it doesn't properly fit with "Customer home tracks events are prefixed with calypso_customer_home...". So, in that case, I wonder if

1) Make sense to create two records for the same event? (🤔)

2) How advisable is compare the stats with events that have different names? If it's doable, we could use straight the `calypso customer home` prefix.

Reply:
cc @martinremy

Reply:
> 1. Make sense to create two records for the same event? (🤔)

This is what you're suggesting here, @gwwar?

> This can also be grouped with the requestInlineHelpSearchResults as a meta property on the action object.